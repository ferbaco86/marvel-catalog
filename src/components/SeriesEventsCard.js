import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
// import { Redirect } from 'react-router-dom';

const imageSize = 'portrait_uncanny';

const Card = styled.div`
display: flex;
flex-direction: column;
align-items: center;
background-color: #202020;
flex: 0 0 20%;
margin: 1rem;
cursor: pointer;
&:hover {
  transform: scale(1.05);
  transition: all .2s linear;
  }
@media screen and (max-width: 1200px) {
    flex: 0 0 30%;
  }
@media screen and (max-width: 800px) {
  width: 100%;
  align-self: center;
  }
}`;

const Title = styled.h1`
color: white;
padding: 1rem;
font-family: 'Roboto Condensed', sans-serif;
font-weight: 700;
text-align: center;

@media screen and (max-width: 800px) {
  width: min-content;
  text-align: center;
  margin: 0 auto;
  }`;

const Pic = styled.img`
border-bottom: 5px solid #ec1d24;
@media screen and (max-width: 800px) {
  width: 100%;
  }
`;

const StyledLink = styled.a`
text-decoration: none;
@media screen and (max-width: 800px) {
  width: 100%;
  }`;

const SeriesEventsCard = props => {
  const {
    name, image, extension, url,
  } = props;

  return (
    <Card>
      <StyledLink href={url}>
        <Pic src={`${image}/${imageSize}.${extension}`} alt={`${name} Portrait`} />
        <Title>{name}</Title>
      </StyledLink>
    </Card>
  );
};

SeriesEventsCard.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  extension: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

export default SeriesEventsCard;
