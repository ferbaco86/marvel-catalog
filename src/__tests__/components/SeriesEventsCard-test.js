import { render, screen, act } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../reducers/index';
import SeriesEventsCard from '../../components/SeriesEventsCard';

const character = {
  name: 'Peter Parker',
  image: 'www.mockurl.com',
  extension: 'jpg',
  url: 'www.mockurl2.com',
};

describe('Rendering component', () => {
  it('creates a SeriesEventsCard component', () => {
    act(() => {
      render(
        <BrowserRouter>
          <Provider store={store}>
            <SeriesEventsCard
              name={character.name}
              image={character.image}
              extension={character.extension}
              url={character.url}
            />
          </Provider>
        </BrowserRouter>,
      );
    });
    screen.getByText('Peter Parker');
  });
});

describe('Display', () => {
  it('renders correctly', () => {
    const snap = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <SeriesEventsCard
            name={character.name}
            image={character.image}
            extension={character.extension}
            url={character.url}
          />
        </Provider>
      </BrowserRouter>,
    ).toJSON();
    expect(snap).toMatchSnapshot();
  });
});
