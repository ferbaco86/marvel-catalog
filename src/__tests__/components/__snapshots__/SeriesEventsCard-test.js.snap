// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display renders correctly 1`] = `
<div
  className="sc-bdfBwQ iEmcFJ"
>
  <a
    className="sc-hKgILt bWnvia"
    href="www.mockurl2.com"
  >
    <img
      alt="Peter Parker Portrait"
      className="sc-dlfnbm ckVWtK"
      src="www.mockurl.com/portrait_uncanny.jpg"
    />
    <h1
      className="sc-gsTCUz kqqFgO"
    >
      Peter Parker
    </h1>
  </a>
</div>
`;
